Move Katello react tests out of parallel steps

johnpmitsch opened the pull request:
With the addition of functional tests that use a lot of asynchronous actions, we are seeing timeout issues for these tests in Jenkins but not in GH actions. Moving them out of the parallel steps should help to address these timeout issues. We have already increased the timeouts for jest, react-testing-lib, and nock.

The goal is to remove the react tests (and foreman npm install step) from the PR job and only keep it in the nightly/release job. This would allow us to rely on GH actions in PRs and speed up the jenkins job. I think it is a good idea to see the tests run successfully for a few days in the new non-parallel step before removing it, so we can be sure there won't be issues during the release jobs.

ehelms commented:
This will also move to more of a fail fast model where React tests, if failing, will fail the build quickly but won't show if other test suites may not be passing. That's more an observation to ensure understanding of the other side of the effects of this. If the Katello project wants to go this route with their testing happy to merge. 

@johnpmitsch do you want to continue with this route or test @ekohl suggestion for a period of time?

johnpmitsch commented:
> This will also move to more of a fail fast model where React tests, if failing, will fail the build quickly but won't show if other test suites may not be passing. That's more an observation to ensure understanding of the other side of the effects of this. If the Katello project wants to go this route with their testing happy to merge.
> 
> @johnpmitsch do you want to continue with this route or test @ekohl suggestion for a period of time? (I think we'd need to use `agent fast` to keep it on the same node set)

We definitely should avoid failing fast on React tests.

I moved the npm install steps for katello and tests together with npm install for foreman as a separate stage and added `agent any`.

johnpmitsch commented:
@ekohl @ehelms should the new stage be inside the parallel steps? That is what makes it non-blocking right? And the agent `any` will ensure it gets separate resources now?

johnpmitsch commented:
> @ekohl @ehelms should the new stage be inside the parallel steps? That is what makes it non-blocking right? And the agent `any` will ensure it gets separate resources now?

Re-reading your comments, I'm pretty sure you meant for the stage to be in the parallel steps. I updated the PR to this, let me know if I've misinterpreted anything!

ekohl commented on puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/source/katello.groovy:
```diff
@@ -48,13 +48,6 @@ pipeline {
 
             }
         }
-        stage('Install Foreman npm packages') {
```
You still need NPM installed for the assets-precompile step

johnpmitsch commented:
Going to close this as the npm install step will be needed for asset precompilation and we have to handle releases, so moving the react tests won't be much improvement while the asset precompilation is in Jenkins. Giving this approach a try instead:

https://github.com/theforeman/foreman-infra/pull/1495
